source/derived.tex

```diff
@@ -910,6 +910,45 @@
 function with a deleted definition.%
 \indextext{function!virtual|)}
 
+\pnum
+If an overriding function specifies contract conditions\iref{dcl.attr.contract},
+it shall specify the same list of contract conditions as
+its overridden functions;
+no diagnostic is required
+if corresponding conditions will always evaluate to the same value.
+Otherwise, it is considered to have
+the list of contract conditions from one of its overridden functions;
+the names in the contract conditions are bound,
+and the semantic constraints are checked,
+at the point where the contract conditions appear.
+Given a virtual function \tcode{f}
+with a contract condition that odr-uses \tcode{*this}\iref{basic.def.odr},
+the class of which \tcode{f} is a direct member
+shall be be an unambiguous and accessible base class of any class
+in which \tcode{f} is overridden.
+If a function overrides more than one function,
+all of the overridden functions shall have
+the same list of contract conditions\iref{dcl.attr.contract};
+no diagnostic is required
+if corresponding conditions will always evaluate to the same value.
+\begin{example}
+\begin{codeblock}
+struct A {
+  virtual void g() [[expects: x == 0]];
+  int x = 42;
+};
+
+int x = 42;
+struct B {
+  virtual void g() [[expects: x == 0]];
+}
+
+struct C : A, B {
+  virtual void g();             // ill-formed, no diagnostic required
```

Comment:
Wait, why is this "no diagnostic required"? The overrider doesn't satisfy the conditions of the ODR, and won't always produce the same value. This is just ill-formed, isn't it?

Reply:
Right, fixed.